### PR TITLE
bounty: BLAKE3 implementation

### PR DESCRIPTION
**Pull request by bjsi**
This is my attempt at the BLAKE3 bounty.

### CODE REVIEW

File: sha3.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+def digest(state: Tensor) -> str:
+  return state.flatten().numpy().tobytes()[:md_len].hex()
+
+
+def tinygrad_sha3_256(message: str) -> str:
+  msg_bytes = bytearray(message.encode("utf-8"))
+  state = Tensor.zeros(height * width, dtype=dtypes.uint64).contiguous()
+
+  # absorb
+  for block_idx in range(0, len(msg_bytes), rate_bytes):
```
Comment (geohot) [inline sha3.py:148]: This is not parallel

Comment (bjsi) [reply sha3.py]: Updated to implement batch-level parallelism.

File: sha3.py [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+def tinygrad_sha3_256(messages: List[str]) -> List[str]:
+  """
+  Hash a batch of equal length messages using SHA3-256.
+  """
+  batch_size = len(messages)
+  states = Tensor.zeros((batch_size, height, width), dtype=dtypes.uint64).contiguous()
+  batched_blocks, msg_bytes = messages_to_blocks(messages)
+  n_blocks = batched_blocks.shape[0]
+  # absorb phase
+  for block_idx in range(0, n_blocks):
```
Comment (geohot) [inline sha3.py:178]: This is not parallel

**Comment by bjsi** [comment]
closing in favour of https://github.com/tinygrad/tinygrad/pull/7773

**Pull request closed without merging**